src/codegen/mod.rs

```diff
@@ -2105,8 +2109,24 @@ impl CodeGenerator for CompInfo {
 
                                         Some(quote! {
                                             assert_eq!(
-                                                unsafe {
-                                                    &(*(::#prefix::ptr::null::<#canonical_ident>())).#field_name as *const _ as usize
+                                                {
+                                                    // Create an instance of #canonical_ident struct from zero bit pattern
+                                                    let struct_instance = unsafe {
+                                                        std::mem::zeroed::<#canonical_ident>()
+                                                    };
+
+                                                    // Get the pointers to the struct and its field
+                                                    let struct_ptr = &struct_instance as *const #canonical_ident;
```

Comment:
This potential UB can be avoided:

```suggestion
                                                    let struct_instance = core::mem::MaybeUninit::<#canonical_ident>::uninit();

                                                    // Get the pointers to the struct and its field
                                                    let struct_ptr = struct_instance.as_ptr();
```